### PR TITLE
Add E2E Tests for UDPRoute

### PR DESCRIPTION
**Pull request by cnvergence**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/envoyproxy/gateway/issues/2011

### CODE REVIEW

File: test/e2e/tests/udproute.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+func parentsForRouteMatch(t *testing.T, routeName types.NamespacedName, expected, actual []gatewayv1.RouteParentStatus, namespaceRequired bool) bool {
+	t.Helper()
+
+	if len(expected) != len(actual) {
+		t.Logf("Route %s/%s expected %d Parents got %d", routeName.Namespace, routeName.Name, len(expected), len(actual))
+		return false
+	}
+
+	// TODO(robscott): Allow for arbitrarily ordered parents
```
Comment (arkodg) [inline test/e2e/tests/udproute.go:162]: rm this TODO

Comment (arkodg) [reply test/e2e/tests/udproute.go:162]: if this code is referenced from upstream, lets add that acknowledgement in the copyright similar to https://github.com/envoyproxy/gateway/blob/8acb1ee0ab375221fb82dc7fe034ef6dded9ac6e/internal/status/status.go#L6

File: tools/linter/codespell/.codespell.ignorewords [hunk 2]
```diff
@@ -1,4 +1,5 @@
 keypair
 keypairs
 als
-requestor
\ No newline at end of file
+requestor
+aParent
```
Comment (arkodg) [inline tools/linter/codespell/.codespell.ignorewords:5]: recommend rming this, and using another name for the var that doesnt confuse `codespell` :)

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**